lib/grandpa/errors.go

```diff
@@ -63,6 +63,8 @@ var (
 	// ErrNoJustification is returned when no justification can be found for a block, ie. it has not been finalised
 	ErrNoJustification = errors.New("no justification found for block")
 
+	ErrBlockHashMismatch = errors.New("block hash does not correspond to given block number")
```

Comment:
nit a tiny bit clearer

```suggestion
	ErrBlockHashMismatch = errors.New("block hash mismatches hash of block for the given block number")
```